admin/registration_schemas/views.py

```diff
@@ -0,0 +1,160 @@
+        return reverse_lazy('registration_schemas:detail', kwargs={'registration_schema_id': self.get_object().id})
+
+
+class RegistrationSchemaCreateView(FormView, PermissionRequiredMixin):
+    """
+    Allows authorized users to create a Registration Schema.
+    """
+    template_name = 'registration_schemas/registration_schema_create.html'
+    permission_required = 'osf.change_registrationschema'
+    raise_exception = True
+    form_class = RegistrationSchemaCreateForm
+
+    def form_valid(self, form):
+        latest_version = RegistrationSchema.objects.filter(
+            name=form.data['name']
+        ).aggregate(latest_version=Max('schema_version'))['latest_version'] or 0
+
+        blocks = self.csv_to_blocks(form.files['schema'])
+        registration_schema = RegistrationSchema.objects.create(
+            name=form.data['name'],
+            schema={
+                'atomicSchema': True,
+                'version': latest_version + 1,
+                'blocks': blocks,
+            },
+            active=False,
+            visible=False,
+            schema_version=latest_version + 1,
+        )
+        create_schema_blocks_for_atomic_schema(registration_schema)
+
+        self.success_url = reverse_lazy(
+            'registration_schemas:detail',
+            kwargs={
+                'registration_schema_id': registration_schema.id,
+            }
+        )
+        messages.success(
+            self.request,
+            f'{registration_schema.name} ({registration_schema.schema_version}) created with '
+            f'{registration_schema.schema_blocks.count()} blocks.'
+        )
+
+        return super().form_valid(form)
+
+    def csv_to_blocks(self, file):
+        blocks = []
+        supported_keys = {field.name for field in RegistrationSchemaBlock._meta.get_fields()}
+
+        for row in csv.DictReader(codecs.iterdecode(file.file, 'utf-8'), delimiter=','):
+            data = {}
+            for key, value in row.items():
+                if key not in supported_keys or not value:
+                    continue
+                if key != 'required':
+                    data[key] = value
+                else:
+                    data[key] = True if value.lower() == 'true' else False
+            blocks.append(data)
+
+        return blocks
+
+
+class RegistrationSchemaDeleteView(DeleteView, PermissionRequiredMixin):
+    """
+    Allows authorized users to delete a Registration Schema.
+    """
+    permission_required = 'osf.change_registrationschema'
+    raise_exception = True
+    form_class = RegistrationSchemaCreateForm
+    success_url = reverse_lazy('registration_schemas:list')
+    model = RegistrationSchema
+
+    def get_object(self, queryset=None):
+        return RegistrationSchema.objects.get(id=self.kwargs['registration_schema_id'])
+
+    def delete(self, request, *args, **kwargs):
+        schema = self.get_object()
+        providers = schema.providers.all()
+        if providers:
```

Comment:
The more I think about it, I don't think we need to check this. As is, it's going to add an extra step to the process when iterating, especially since we discussed optionally copying providers as a follow-up.

Reply:
agreed, the registration check does enough.